native/src/boot/main.cpp

```diff
@@ -142,6 +145,21 @@ int main(int argc, char *argv[]) {
         unlink(EXTRA_FILE);
         unlink(RECV_DTBO_FILE);
         unlink(DTB_FILE);
+        unlink(BOOTCONFIG_FILE);
+
+        char file_namen_pattern[sizeof(VENDOR_RAMDISK_FILE)];
+        ssprintf(file_namen_pattern, sizeof(file_namen_pattern), VENDOR_RAMDISK_FILE, 3, ",");
+        string file_name(file_namen_pattern);
+        size_t split_pos = file_name.find(",");
+        string start_match = file_name.substr(0, split_pos);
+        string end_match = file_name.substr(split_pos + 1);
```

Comment:
ugly. `VENDOR_RAMDISK_FILE` is something at compile time. why deal with it at runtime?

Reply:
replaced with compile time constants